art/estimators/certification/derandomized_smoothing/derandomized_smoothing.py

```diff
@@ -0,0 +1,445 @@
+                 model if a position is ablated.
+        """
+        if not self.channels_first:
+            x = np.transpose(x, (0, 3, 1, 2))
+
+        if row_pos is None:
+            row_pos = random.randint(0, x.shape[2])
+        if column_pos is None:
+            column_pos = random.randint(0, x.shape[3])
+
+        x = np.concatenate([x, 1.0 - x], axis=1)
+
+        if isinstance(row_pos, list) and isinstance(column_pos, list):
+            for i, (row, col) in enumerate(zip(row_pos, column_pos)):
+                x[i : i + 1] = self.ablate(x[i : i + 1], row_pos=row, column_pos=col)
+        elif isinstance(row_pos, int) and isinstance(column_pos, int):
+            x = self.ablate(x, row_pos=row_pos, column_pos=column_pos)
+
+        if not self.channels_first:
+            x = np.transpose(x, (0, 2, 3, 1))
+        return x
+
+    def ablate(self, x: np.ndarray, column_pos: int, row_pos: int) -> np.ndarray:
+        """
+        Ablates the image only retaining a block starting at (row_pos, column_pos) of height/width "self.ablation_size"
+
+        :param x: input data
+        :param row_pos: Specifies the row index where to retain the image block.
+        :param column_pos: Specifies the column index where to retain the image block.
+        :return: Data ablated at all locations aside from the specified block.
+        """
+        k = self.ablation_size
+        num_of_image_columns = x.shape[3]
+        num_of_image_rows = x.shape[2]
+
+        if row_pos + k > x.shape[2] and column_pos + k > x.shape[3]:
+            start_of_ablation = column_pos + k - num_of_image_columns
+            x[:, :, :, start_of_ablation:column_pos] = 0.0
+
+            start_of_ablation = row_pos + k - num_of_image_rows
+            x[:, :, start_of_ablation:row_pos, :] = 0.0
+
+        # only the row wraps
+        elif row_pos + k > x.shape[2] and column_pos + k <= x.shape[3]:
+            x[:, :, :, :column_pos] = 0.0
+            x[:, :, :, column_pos + k :] = 0.0
+
+            start_of_ablation = row_pos + k - num_of_image_rows
+            x[:, :, start_of_ablation:row_pos, :] = 0.0
+
+        # only column wraps
+        elif row_pos + k <= x.shape[2] and column_pos + k > x.shape[3]:
+            start_of_ablation = column_pos + k - num_of_image_columns
+            x[:, :, :, start_of_ablation:column_pos] = 0.0
+
+            x[:, :, :row_pos, :] = 0.0
+            x[:, :, row_pos + k :, :] = 0.0
+
+        # neither wraps
+        elif row_pos + k <= x.shape[2] and column_pos + k <= x.shape[3]:
+            x[:, :, :, :column_pos] = 0.0
+            x[:, :, :, column_pos + k :] = 0.0
+
+            x[:, :, :row_pos, :] = 0.0
+            x[:, :, row_pos + k :, :] = 0.0
+        else:
+            raise ValueError(f"Ablation failed on row: {row_pos} and column: {column_pos} with size {k}")
+
+        return x
+
+
+ABLATOR_TYPE = Union[BlockAblator, ColumnAblator]  # pylint: disable=C0103
```

Comment:
Is it possible to move this type to `art.utils` to the other type definitions?

Reply:
I agree, moved it!